File: csharp/rna-transcription/Iteration1/RnaTranscription.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class RnaTranscription
{
    private static Dictionary<char, char> dnaToRnaMap = new()
    {
        { 'G', 'C' },
        { 'C', 'G' },
        { 'T', 'A' },
        { 'A', 'U' }
    };
    
    public static string ToRna(string nucleotide) =>
        string.Concat(nucleotide.Select(dna => dnaToRnaMap[dna]));
}

File: c/rna-transcription/Iteration2/rna_transcription.c
#include "rna_transcription.h"
#include <string.h>
#include <stdlib.h>

static char map_nucleotide(char nucleotide) {
    switch (nucleotide) {
        case 'G': return 'C';
        case 'C': return 'G';
        case 'T': return 'A';
        case 'A': return 'U';
        default: return nucleotide;
    }
}

char *to_rna(const char *dna) {
    int len = strlen(dna);
    char *rna = malloc(len * sizeof(char));

    for (int i = 0; i < len; i++) {
        rna[i] = map_nucleotide(dna[i]);
    }

    return rna;
}
